### PR TITLE
Migrate watchlist to postgres

### PR DESCRIPTION
**Pull request by cole-adams**
Changes:
* Migrate watchlist to cockroach using list table
* Add table for next work
* changed search function to use new types (and added mapping function to still support old review type)

### CODE REVIEW

File: src/features/reviews/components/AddReviewPrompt.vue [hunk 1]
```diff
@@ -19,32 +19,46 @@ import { useRoute } from "vue-router";
...
 }>();
 
 const route = useRoute();
-
-const { data: watchList, isLoading: watchListLoading } = useWatchList(
-  route.params.clubId as string
+const clubId = useClubId();
+const { data: watchList, isLoading: watchListLoading } = useList(
+  clubId,
+  "watchlist"
```
Comment (brian-norman) [inline src/features/reviews/components/AddReviewPrompt.vue:34]: We should expose the enum types we created in the DB as the types of lists to the frontend so we don't need to pass these strings around

File: src/features/reviews/components/AddReviewPrompt.vue [hunk 2]
```diff
@@ -19,32 +19,46 @@ import { useRoute } from "vue-router";
...
+          poster_path: item.imageUrl ?? "",
+        }
+    ) ?? []
 );
 
-const { mutate: deleteMovie, isLoading: deleteLoading } = useDeleteMovie(
-  route.params.clubId as string
-);
+const { mutate: deleteWatchlistItem, isLoading: deleteLoading } =
+  useDeleteListItem(clubId, "watchlist");
```
Comment (brian-norman) [inline src/features/reviews/components/AddReviewPrompt.vue:51]: Same here about using the enum

File: src/features/watch-list/components/ClubBacklog.vue [hunk 3]
```diff
@@ -48,67 +48,40 @@
...
 const { searchTerm, clearSearch } = defineProps<{
   searchTerm: string;
   clearSearch: () => void;
 }>();
 
-const route = useRoute();
-
 const clubId = useClubId();
-const { data } = useWatchList(route.params.clubId as string);
+const { data: watchList } = useList(clubId, "watchlist");
```
Comment (brian-norman) [inline src/features/watch-list/components/ClubBacklog.vue:67]: Use an enum

File: src/features/watch-list/components/ClubBacklog.vue [hunk 4]
```diff
@@ -48,67 +48,40 @@
...
 const { data: backlog } = useList(clubId, "backlog");
-const watchList = computed(() => (data.value ? data.value.watchList : []));
 
 const { mutate: deleteBacklogItem } = useDeleteListItem(clubId, "backlog");
-const { mutate: addMovie } = useAddMovie(route.params.clubId as string);
+const { mutate: addToWatchlist } = useAddListItem(clubId, "watchlist");
```
Comment (brian-norman) [inline src/features/watch-list/components/ClubBacklog.vue:71]: Use an enum

File: src/features/watch-list/components/WatchList.vue [hunk 5]
```diff
@@ -58,55 +57,57 @@ const { searchTerm, clearSearch } = defineProps<{
 const route = useRoute();
 const router = useRouter();
 
-const { mutate: deleteMovie } = useDeleteMovie(route.params.clubId as string);
+const clubId = useClubId();
+const { mutate: deleteWatchlistItem } = useDeleteListItem(clubId, "watchlist");
```
Comment (brian-norman) [inline src/features/watch-list/components/WatchList.vue:61]: Enum :)

File: src/features/watch-list/components/WatchList.vue [hunk 6]
```diff
@@ -58,55 +57,57 @@ const { searchTerm, clearSearch } = defineProps<{
...
-  if (nextMovieItem) {
-    const sortedWatchList = watchList.value.filter(
-      (movie) => movie.movieId !== nextMovieItem.movieId
-    );
-    sortedWatchList.unshift(nextMovieItem);
-    return sortedWatchList;
-  }
-  return watchList.value;
-});
+const { data: watchList } = useList(clubId, "watchlist");
```
Comment (brian-norman) [inline src/features/watch-list/components/WatchList.vue:71]: enum

File: src/features/watch-list/views/WatchListView.vue [hunk 7]
```diff
@@ -44,19 +41,18 @@
...
-const route = useRoute();
+import { useClubId } from "@/service/useClub";
+import { useList } from "@/service/useList";
 
 const searchInput = ref<HTMLInputElement | null>(null);
 const searchInputSlash = ref<HTMLParagraphElement | null>(null);
 
-const { isLoading } = useWatchList(route.params.clubId as string);
+const clubId = useClubId();
+const { isLoading } = useList(clubId, "watchlist");
```
Comment (brian-norman) [inline src/features/watch-list/views/WatchListView.vue:55]: enum

File: src/features/reviews/components/AddReviewPrompt.vue (hunk 1 above)
Comment (cole-adams) [reply src/features/reviews/components/AddReviewPrompt.vue]: Good point. I've updated all the `list` hooks to only accept `WorkListType` and updated all instances so we should get type errors now if we don't use the enum

**Review by brian-norman — APPROVED** [review]

**Pull request merged**